src/consensus/read_validator.rs

```diff
@@ -96,7 +98,41 @@ impl ReadValidator {
         }
     }
 
+    fn verify_signatures(value: &proto::DecidedValue) -> bool {
+        let certificate = match value.value.as_ref().unwrap() {
+            proto::decided_value::Value::Shard(shard_chunk) => shard_chunk
+                .commits
+                .as_ref()
+                .unwrap()
+                .to_commit_certificate(),
+
+            proto::decided_value::Value::Block(block) => {
+                block.commits.as_ref().unwrap().to_commit_certificate()
+            }
+        };
+
+        for signature in certificate.aggregated_signature.signatures {
```

Comment:
Wondering if 0 signatures probably should be treated as invalid too?

Reply:
Yeah, it should be invalid